spec/features/idv/steps/review_step_spec.rb

```diff
@@ -0,0 +1,113 @@
+feature 'idv review step', :idv_job do
+  include IdvStepHelper
+
+  it 'requires the user to enter the correct password to redirect to confirmation step' do
+    start_idv_from_sp
+    complete_idv_steps_before_review_step
+
+    fill_in 'Password', with: 'this is not the right password'
+    click_idv_continue
+
+    expect(page).to have_content(t('idv.errors.incorrect_password'))
+    expect(page).to have_current_path(verify_review_path)
+
+    fill_in 'Password', with: user_password
+    click_idv_continue
+
+    expect(page).to have_content(t('headings.personal_key'))
+    expect(page).to have_current_path(verify_confirmations_path)
+  end
+
+  it 'displays the profile information for the user' do
+    start_idv_from_sp
+    complete_idv_steps_before_review_step
+
+    expect(page).to have_content('José')
+    expect(page).to have_content('One')
+    expect(page).to have_content('123 Main St')
+    expect(page).to have_content('Nowhere, VA 6604')
+    expect(page).to have_content('January 02, 1980')
+    expect(page).to have_content('666-66-1234')
+    expect(page).to have_content('+1 (555) 555-0000')
+  end
+
+  context 'choosing to confirm address with phone' do
+    let(:user) { user_with_2fa }
+
+    before do
+      start_idv_from_sp
+      complete_idv_steps_with_phone_before_review_step(user)
+    end
+
+    it 'does not send a letter and creates a verified profile' do
+      fill_in 'Password', with: user_password
+      click_idv_continue
+
+      expect(user.events.account_verified.size).to be(1)
+      expect(user.profiles.count).to eq 1
+
+      profile = user.profiles.first
+
+      expect(profile.active?).to eq true
+      expect(profile.phone_confirmed).to eq true
+      expect(UspsConfirmation.count).to eq(0)
+    end
+  end
+
+  context 'choosing to confirm address with usps' do
+    let(:user) { user_with_2fa }
+    let(:sp) { :oidc }
+
+    before do
+      start_idv_from_sp(sp)
+      complete_idv_steps_with_usps_before_review_step(user)
+    end
+
+    it 'sends a letter and creates an unverified profile' do
+      fill_in 'Password', with: user_password
+
+      expect { click_continue }.
+        to change { UspsConfirmation.count }.from(0).to(1)
+
+      expect(user.events.account_verified.size).to be(0)
+      expect(user.profiles.count).to eq 1
+
+      profile = user.profiles.first
+
+      expect(profile.active?).to eq false
+      expect(profile.phone_confirmed).to eq false
+    end
+
+    context 'with an sp' do
+      it 'sends a letter with a reference the sp' do
+        fill_in 'Password', with: user_password
+        click_continue
+
+        usps_confirmation_entry = UspsConfirmation.last.decrypted_entry
+
+        if sp == :saml
+          expect(usps_confirmation_entry.issuer).
+            to eq('https://rp1.serviceprovider.com/auth/saml/metadata')
+        else
+          expect(usps_confirmation_entry.issuer).
+            to eq('urn:gov:gsa:openidconnect:sp:server')
+        end
+      end
+    end
```

Comment:
Can we combine this expectation with the previous one since they're both coming from an SP?

Reply:
I was going to suggest that we unit test this instead but as I started looking at the code, I noticed it's quite gnarly and misleading. The first thing the `create` action in `UspsController` does is `create_user_event(:usps_mail_sent, current_user)`, which is not true, and on the `verify/review` page, the user sees a flash message that says "Your letter is on its way", which is not true either. The letter doesn't get sent until after you enter your password on the `/verify/review` page, and that is hidden inside a method called `init_profile`, which calls `idv_session.complete_session`, which is the one that actually generates the letter by calling `create_usps_entry`. We should clean this up.

Reply:
Yeah, my eventual goal is to be able to come back to tear a lot of these down and replace them with unit tests. But before we can make those refactors, we'll want to have good coverage at the feature level. That way we can be confident our changes aren't leading to strange or inconsistent behavior, and we don't find ourselves in the place we are with the something like the cancel behavior.

Reply:
Also, re combining this with the previous one, I think that makes it a lot harder to track down where this logic is being tested which means that when we do fix it, it will be harder to figure out what we need to come back and tear down, especially as that spec changes over time.